### PR TITLE
Last ETag support in MetadataClient

### PR DESCRIPTION
**Pull request by evildour**

### CODE REVIEW

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/MetadataClientImpl.cs [hunk 1]
```diff
@@ -91,6 +92,17 @@ public MetadataClientImpl(IHttpClientFactory httpClientFactory)
         private static string CreateAbsoluteUrl(string relativeUrl) =>
             $"http://{MetadataHost.Value}/computeMetadata/v1/{relativeUrl}";
 
+        private static string GetETag(HttpResponseMessage response)
+        {
+            IEnumerable<string> eTagValues;
+            if (response.Headers.TryGetValues("ETag", out eTagValues))
```
Comment (jskeet) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/MetadataClientImpl.cs]: You can just use:

```csharp
response.Headers.TryGetValues("ETag", out eTagValues);
return eTagValues?.FirstOrDefault();
```

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/WaitForChangeResult.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+// limitations under the License.
+
+namespace Google.Cloud.Metadata.V1
+{
+    /// <summary>
+    /// Contains the result from the wait for change operation.
+    /// </summary>
+    /// <seealso cref="MetadataClient.WaitForChange"/>
+    /// <seealso cref="MetadataClient.WaitForChangeAsync"/>
+    public class WaitForChangeResult
```
Comment (jskeet) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/WaitForChangeResult.cs]: Seal the class?

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/MetadataClientImpl.cs (hunk 1 above)
Comment (evildour) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/MetadataClientImpl.cs]: Done

**Comment by evildour** [comment]
(rebased to pick up other metadata changes)

**Pull request merged**